ethereum/contracts/state-transition/chain-deps/facets/Admin.sol

```diff
@@ -105,7 +112,21 @@ contract AdminFacet is Base, IAdmin {
 
     /// @notice Instantly pause the functionality of all freezable facets & their selectors
     /// @dev Only the governance mechanism may freeze Diamond Proxy
-    function freezeDiamond() external onlyGovernor {
+    function freezeNotUpdated(uint256 _protocolVersion) external onlyStateTransition {
+        Diamond.DiamondStorage storage diamondStorage = Diamond.getDiamondStorage();
+
+        if (chainStorage.protocolVersion == _protocolVersion) {
```

Comment:
I would probably suggest allowing to have `chainStorage.protocolVersion >= _protocolVersion`, in case a bug is found in some rare old version

Reply:
We don't need this anymore, since we check protocol version when executing, if the protocol is wrong we cannot update.

Reply:
We might want an emergency freeze however, but that is a security feature and not related to the core upgrade mechanism.